cpp/oneapi/dal/algo/knn/common.hpp

```diff
@@ -22,27 +22,61 @@
 namespace oneapi::dal::knn {
 
 namespace task {
+namespace v1 {
 struct classification {};
 using by_default = classification;
+} // namespace v1
+
+using v1::classification;
+using v1::by_default;
+
 } // namespace task
 
+namespace method {
+namespace v1 {
+struct kd_tree {};
+struct brute_force {};
+using by_default = brute_force;
+} // namespace v1
+
+using v1::kd_tree;
+using v1::brute_force;
+using v1::by_default;
+
+} // namespace method
+
 namespace detail {
+namespace v1 {
 struct tag {};
-
-template <typename Task = task::by_default>
+template <typename Task>
 class descriptor_impl;
 
+template <typename Task>
 class model_impl;
+} // namespace v1
+
+using v1::tag;
+using v1::descriptor_impl;
+using v1::model_impl;
+
+template <typename Float>
+constexpr bool is_valid_float_v = dal::detail::is_one_of_v<Float, float, double>;
```

Comment:
Is it specific for kNN?

Reply:
There is uncertainty in floating point types need to be supported in the future, this could be different for different algorithms, so I've made algorithm-specific `is_valid_float`